crypto/utils/utils.go

```diff
@@ -398,3 +397,12 @@ func ExtendHashOutput(salt, message []byte, outputByteLength int) []byte {
 	}
 	return result
 }
+
+func Pad(x []byte, n int) []byte {
+	pad := make([]byte, n-len(x))
+	return append(pad, x...)
+}
+
+func Bytes32(x *big.Int) []byte {
```

Comment:
`func Bytes32(x *big.Int) [32]byte`

Reply:
if the output is [32]byte, then we can not use "append".